include/Nazara/Utility/Algorithm.hpp

```diff
@@ -57,6 +60,49 @@ namespace Nz
 	NAZARA_UTILITY_API void SkinPositionNormalTangent(const SkinningData& data, unsigned int startVertex, unsigned int vertexCount);
 
 	NAZARA_UTILITY_API void TransformVertices(VertexPointers vertexPointers, unsigned int vertexCount, const Matrix4f& matrix);
+	
+
+	template <typename T, std::enable_if_t<std::is_same<std::remove_const_t<T>, Color>::value>...>
+	ComponentType ComponentTypeOf() { return ComponentType_Color; }
```

Comment:
I would prefer it named "GetComponentTypeOf"

Reply:
Also, it should have a default implementation with a static_assert(false) saying something like "missing implementation for type"